src/main/java/com/ericsson/ei/subscription/SubscriptionValidator.java

```diff
@@ -70,18 +75,22 @@ public static void validateSubscription(Subscription subscription) throws Subscr
     }
 
     /**
-     * Validation of subscriptionName parameter Throws SubscriptionValidationException if validation of
-     * the parameter fails due to wrong format of parameter.
+     * Validation of subscriptionName parameter Throws
+     * SubscriptionValidationException if validation of the parameter fails due to
+     * wrong format of parameter.
      *
      * @param subscriptionName
      */
     private static void validateSubscriptionName(String subscriptionName) throws SubscriptionValidationException {
-        // When this regExp need to be changed then remember to change the one in the
-        // back-end (invalidSubscriptionNameRegex in subscription.js), which do the same
-        // invalid subscription name check. The two
-        // regEx always need to be the same for ensuring the same check.
-        // /(\W)/ Is a regEx that matches anything that is not [A-Z,a-z,0-8] and _.
-        String invalidSubscriptionNameRegex = "(\\W)";
+        // Here the regEx used is :/(\W)/ that matches anything that is not
+        // [A-Z,a-z,0-8] and _.
+        String invalidSubscriptionNameRegex = null;
```

Comment:
What happens if the next try catch fails and this is null in the validation?

Reply:
In the documentation it says that if no mapping is found a NullPointerException is thrown. So you can either catch and log that or you can return a default value like this "getString("invalidName","") and check that it is not empty in an else if. You still need to log something if it can't validate though so perhaps catching the null pointer is best.

Reply:
It is handled now